File: Exception/ErrorBoundary.py
import asyncio


async def resolve_task(value):
    return value


async def reject_task(message):
    raise Exception(message)


async def run_all_tasks():
    tasks = [
        resolve_task("Task 1 succeeded"),
        reject_task("Task 2 failed"),
        resolve_task("Task 3 succeeded"),
        reject_task("Task 4 network error"),
    ]
    results = await asyncio.gather(*tasks, return_exceptions=True)

    for i, result in enumerate(results, 1):
        if isinstance(result, Exception):
            print(f"[X][{i}]: {result}")
        else:
            print(f"[OK][{i}]: {result}")


asyncio.run(run_all_tasks())

File: Exception/ErrorBoundary.js
const tasks = [
  Promise.resolve("Task 1 succeeded"),
  Promise.reject(new Error("Task 2 failed")),
  Promise.resolve("Task 3 succeeded"),
  Promise.reject(new Error("Task 4 network error")),
];

async function runAllTasks() {
  const results = await Promise.allSettled(tasks);

  results.forEach((result, i) => {
    if (result.status === "fulfilled") {
      console.log(`✔ [${i + 1}]: ${result.value}`);
    } else {
      console.log(`✘ [${i + 1}]: ${result.reason.message}`);
    }
  });
}

runAllTasks();
// ✔ [1]: Task 1 succeeded
// ✘ [2]: Task 2 failed
// ✔ [3]: Task 3 succeeded
// ✘ [4]: Task 4 network error